### ISSUE TITLE
WCxf import not working

### ISSUE THREAD

**Issue by peterstangl** [eos/eos#335]
I am getting an error if I try to load Wilson coefficients from a WCxf object (using the current development version of eos). This can be reproduced e.g. by the following code:
```python
import eos
from wilson import Wilson

w = Wilson({'cbtaunutau::cSL': 0.3,}, scale=4.8, eft="WET", basis="EOS")
eos.Parameters.FromWCxf(w.wc)
```
 The error message is
```python
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-2-991fa46dbddc> in <module>
      3 
      4 w = Wilson({'cbtaunutau::cSL': 0.3,}, scale=4.8, eft="WET", basis="EOS")
----> 5 eos.Parameters.FromWCxf(w.wc)

~/.local/lib/python3.8/site-packages/eos/parameter.py in FromWCxf(wc)
    109         parameters = _Parameters.Defaults()
    110         for name in wc.dict:
--> 111             qn     = eos.QualifiedName(name)
    112             prefix = qn.prefix_part()
    113             coeff  = qn.name_part()

NameError: name 'eos' is not defined
```
I guess this can be fixed by simply replacing
https://github.com/eos/eos/blob/edce93909e4632fb646876063002ec54a5de17e0/python/eos/parameter.py#L111
by
```python
            qn     = QualifiedName(name)
```

**Comment by mreboud** [comment]
Thanks @peterstangl. I can reproduce your error with a different Wilson coefficient. For `'cbtaunutau::cSL'` you are also using the development version of wcxf, right?

**Comment by dvandyk** [comment]
Hi @peterstangl,

this error appears, since the wcxf bases are not versioned. I had to start at one end :-)
wcxf-import with the basis recently merged will be working again with the next release.

**Comment by peterstangl** [comment]
@mreboud @dvandyk yes, sorry it maybe was not the best example since I was using the development version of wcxf as well and the basis differs from the released one. But this error is independent of the Wilson coefficient and should be reproducible with any Wilson coefficients no matter which version of the basis is used. The problem is that `eos.QualifiedName` cannot be found since `eos` is not available in the namespace. But `QualifiedName` actually is available since it is imported at the beginning of the file, so changing line 111 as I wrote above should solve this problem.

**Comment by mreboud** [comment]
@peterstangl it indeed solves this problem, but non EOS-based Wilson coefficients don't follow QualifiedName requirements (e.g. they don't have `::`). So I can't really check that your error is the only one for the moment.

**Comment by peterstangl** [comment]
@mreboud yes, there are indeed some other issues:
- when loading the wcxf instance `wc`, it should be checked that `wc.basis == 'EOS'` since the implementation only works with the EOS basis (one can use the wilson package to translate from other bases to the EOS basis).
- the value of `wc.scale` should be checked and only the scale that is used by EOS should be accepted (again the wilson package can be used to run the coefficients to a different scale)

But these two points are independent of the problem described here. Should I open another issue for them?

**Comment by dvandyk** [comment]
Thanks Peter! Nope, we don't need another issue. We can handle all of that in one fell swoop.

**Comment by dvandyk** [comment]
> Hi @peterstangl,
> 
> this error appears, since the wcxf bases are not versioned. I had to start at one end :-)
> wcxf-import with the basis recently merged will be working again with the next release.

Sorry, I was too fast this morning. I had an error recently in the import function related to the change of the basis.
You error is of course unrelated to the basis.

**Comment by dvandyk** [comment]
I have no idea why the 'resolves #335' command in the git commit message did not work. But this has been resolved 10 days ago!
@peterstangl please confirm and close, or holler loudly!

**Comment by peterstangl** [comment]
@dvandyk your commit https://github.com/eos/eos/commit/e8a76ee62bb46dbeaaddc0a927212e1b3e1e5616 seems to be not in the eos repository. This should explain why GitHub did not resolve the issue (the commit is not merged yet) and why I cannot test your changes (which unfortunately I noticed only after recompiling everything :sweat_smile:). But the commit looks good to me. Just one thing: running and matching in wilson often produces numerically small imaginary parts of order 1e-16, which are of course completely negligible. But in this case 
```python
if (not value.imag == 0):
    raise ValueError('WC {0} does not support an imaginary part at present'.format(name))
```
will raise the ValueError. So it might be better to do this only for sizable imaginary parts, e.g.
```python
if abs(value.imag) > 1e-15
```
